File: 2947-count-beautiful-substrings-i/2947-count-beautiful-substrings-i.py
class Solution:
    def beautifulSubstrings(self, s: str, k: int) -> int:
        n = len(s)
        res = 0
        cc = ['a','e','i','o','u']
        for j in range(n):
            c,v=0,0
            for i in range(j,n):
                if s[i] in cc:
                    v += 1
                else:
                    c += 1
                if c==v and (c*v)%k==0:
                    res += 1
        return res

File: 2947-count-beautiful-substrings-i/2947-count-beautiful-substrings-i.java
class Solution {
    public int beautifulSubstrings(String s, int k) {
        int res = 0;
        Set<Character> vowels = new HashSet<>(Set.of('a', 'e', 'i', 'o', 'u'));
        
        for(int i = 0; i < s.length(); i++){
            int countC = 0;
            int countV = 0;
            for(int j = i; j < s.length(); j++){
                char c = s.charAt(j);
                if(vowels.contains(c)){
                    countV++;
                }else{
                    countC++;
                }
                if((countC == countV) && (countC*countV)%k == 0){
                    res++;
                }
            }
        }
        
        return res;
    }
}